src/main/java/frc/robot/RobotContainer.java

```diff
@@ -166,7 +173,15 @@ private void configureButtonBindings() {
                 double yPower = driveController.getLeftPower();
                 double angularPower = driveController.getRotatePower();
                 boolean relative = driveController.getSwerveRelative();
-                swerveSubsystem.setDrivePowers(xPower, yPower, angularPower, relative);
+
+                if (driveController.getHeadingLock()) {
+                    double currentHeading = swerveSubsystem.getRobotPosition().getRotation().getRadians();
+                    double lockHeading = (Math.abs(MathUtil.angleModulus(currentHeading)) > (Math.PI / 2)) ? Math.PI : 0;
```

Comment:
Somewhat confident that `currentHeading` is aways wrapped too, at least according to shuffleboard values when testing the field?